photon-ml/src/main/scala/com/linkedin/photon/ml/PhotonMLCmdLineParser.scala

```diff
@@ -160,9 +162,12 @@ object PhotonMLCmdLineParser {
                       s"Default: ${defaultParams.dataValidationType}")
               .foreach(x => params.dataValidationType = DataValidationType.withName(x.toUpperCase()))
       opt[Boolean](TRAINING_DIAGNOSTICS)
-              .text(s"Whether or not to enable training diagnostics. These can be slow but are very informative. " +
-                      s"Only relevant if you provide a validation set.")
-              .foreach(x => params.trainingDiagnosticsEnabled = x)
+              .text(s"DEPRECATED -- USE $DIAGNOSTIC_MODE")
+              .foreach(x => params.diagnosticMode = if (x) DiagnosticMode.ALL else DiagnosticMode.NONE)
+      opt[String](DIAGNOSTIC_MODE)
+              .text(s"Diagnostic mode after model training (${DiagnosticMode.ALL}, ${DiagnosticMode.NONE}," +
+                  s" ${DiagnosticMode.TRAIN}, ${DiagnosticMode.VALIDATE}). Default: ${defaultParams.diagnosticMode}")
+              .foreach(x => params.diagnosticMode = DiagnosticMode.withName(x.toUpperCase()))
```

Comment:
What's the behaviour if a user specifies both TRAINING_DIAGNOSTICS and DIAGNOSTIC_MODE?